### PR TITLE
Partly update fields with PATCH requests

### PR DESCRIPTION
**Pull request by ActiveChooN**
### Motivation and context

This PR provide ability to patch only changed fields in cvat-core `job` and `task` entities 

### How has this been tested?

Manually

### Checklist

- [x] I submit my changes into the `develop` branch
- [ ] I have added description of my changes into [CHANGELOG](https://github.com/opencv/cvat/blob/develop/CHANGELOG.md) file
~~- [ ] I have updated the [documentation](
  https://github.com/opencv/cvat/blob/develop/README.md#documentation) accordingly~~
~~- [ ] I have added tests to cover my changes~~
~~- [ ] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))~~
- [ ] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [x] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2020 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

File: cvat-core/src/session.js [hunk 1]
```diff
@@ -674,6 +674,11 @@
                 task: undefined,
             };
 
+            let dirtyFields = {
```
Comment (bsekachev) [inline cvat-core/src/session.js:677]: Let's just call it ``updated`` (according to ObjectState which is actually also uses dictionary of updated fields)

File: cvat-core/src/session.js [hunk 2]
```diff
@@ -879,6 +892,13 @@
                 use_cache: undefined,
             };
 
+            let dirtyFields = {
```
Comment (bsekachev) [inline cvat-core/src/session.js:895]: The same comment

File: cvat-core/src/session.js [hunk 3]
```diff
@@ -1443,12 +1473,30 @@
     Job.prototype.save.implementation = async function () {
         // TODO: Add ability to change an assignee
         if (this.id) {
-            const jobData = {
-                status: this.status,
-                assignee_id: this.assignee ? this.assignee.id : null,
-            };
+            const jobData = {};
+
+            for (const [field, isDirty] of Object.entries(this.__dirtyFields)) {
```
Comment (bsekachev) [inline cvat-core/src/session.js:1478]: field, isUpdated

File: cvat-core/src/session.js (hunk 2 above)
Comment (ActiveChooN) [reply cvat-core/src/session.js]: @bsekachev done

**Review by bsekachev — APPROVED** [review]

**Pull request merged**